src/link/Dwarf.zig

```diff
@@ -1095,6 +1097,97 @@ fn addDbgInfoType(
                 try dbg_info_buffer.append(0);
             }
         },
+        .ErrorSet => {
+            // DW.AT.enumeration_type
+            try dbg_info_buffer.append(abbrev_enum_type);
+            // DW.AT.byte_size, DW.FORM.sdata
+            const abi_size = ty.abiSize(target);
+            try leb128.writeULEB128(dbg_info_buffer.writer(), abi_size);
+            // DW.AT.name, DW.FORM.string
+            const name = try ty.nameAllocArena(arena, target);
+            try dbg_info_buffer.writer().print("{s}\x00", .{name});
+
+            // DW.AT.enumerator
+            const no_error = "(no error)";
+            try dbg_info_buffer.ensureUnusedCapacity(no_error.len + 2 + @sizeOf(u64));
+            dbg_info_buffer.appendAssumeCapacity(abbrev_enum_variant);
+            // DW.AT.name, DW.FORM.string
+            dbg_info_buffer.appendSliceAssumeCapacity(no_error);
+            dbg_info_buffer.appendAssumeCapacity(0);
+            // DW.AT.const_value, DW.FORM.data8
+            mem.writeInt(u64, dbg_info_buffer.addManyAsArrayAssumeCapacity(8), 0, target_endian);
+
+            const error_names = blk: {
+                if (ty.isAnyError())
+                    break :blk module.error_name_list.items;
```

Comment:
The problem here is is that the list is not complete until flush(), however updateDecl() and updateFunc() will be called many times with references to this type before it is complete.

Here is my counter-proposal:
 * For debug info purposes, treat all error sets as `anyerror`.
 * Populate the DWARF info for `anyerror` in flush().

This is simple, easy to get right, and provides a completely reasonable debugging experience.

Reply:
OK, I think I've arrived at something I'm pretty happy with. tl;dr we generate `anyerror` in `flush` for errors that were either an `anyerror` or an inferred error set which wasn't resolved yet. From my local testing, works like a charm. This actually brings another topic on the agenda: DWARF testing! Gotta figure out a good test harness given how feature-full it is becoming.